pawnShop/src/main/java/com/example/pawnShop/Controller/ProductTypeController.java

```diff
@@ -0,0 +1,48 @@
+package com.example.pawnShop.Controller;
+
+import com.example.pawnShop.Dto.Product.ProductTypeDto;
+import com.example.pawnShop.Service.Contract.ProductTypeService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/product_type")
+public class ProductTypeController {
+    private final ProductTypeService productTypeService;
+
+    @GetMapping()
+    public ResponseEntity<List<ProductTypeDto>> getAllProductsTypes(){
+        List<ProductTypeDto> productsTypes = productTypeService.getAllProductsTypes();
+
+        return ResponseEntity.ok(productsTypes);
+    }
+    @GetMapping("/{id}")
+    public ResponseEntity<ProductTypeDto> getProductTypeById(@PathVariable UUID id){
+        ProductTypeDto productType = productTypeService.getProductTypeById(id);
+
+        return ResponseEntity.ok(productType);
+    }
+    @PutMapping("/{newProductTypeName}")
```

Comment:
Here you don't need this param. You need Id of entity as parameter and @Requestbody annotation. Check here https://www.baeldung.com/http-put-patch-difference-spring

Reply:
https://www.javaguides.net/2023/07/patchmapping-spring-boot-example.html?m=1&utm_content=cmp-true 


Here you can read it too